README.md

```diff
@@ -0,0 +1,265 @@
+- [ ] Use a simple authorization scheme.
+
+### Client
+
+- [ ] CLI should be able to connect to worker service and start, stop, get
+      status, and stream output of a job.
+
+## Repository Organization & Conventions
+
+The repository follows
+[standard Go conventions on module organization](https://go.dev/doc/modules/layout).
+
+```sh
+tp-worker/
+  go.mod
+  go.sum
+  cmd/
+    tp-worker-server/
+      tp-worker-server.go
+    tp-worker-client/
+      tp-worker-client.go
+  internal/
+    authn/
+      authn.go
+      authn_test.go
+    authz/
+      authz.go
+      authz_test.go
+    job/
+      job.go
+      job_test.go
+    output-storage/
+      output-storage.go
+      output-storage_test.go
+  job/
+    job.go
+    job_test.go
+  proto/
+    tp-worker.proto
+```
+
+As much as possible, following [Effective Go](https://go.dev/doc/effective_go)
+coding conventions will be adhered to. This should help in avoiding some common
+pitfalls such as data races, crashes, concurrency issues, deadlocks, etc.
+
+Testing will focus primarily on happy path, with added tests for intentional
+error conditions, such as when a command is not found, output is not found, user
+doesn't have authorization, etc. Extensive mocking, table-driven tests, fuzzing,
+and mutation testing could be added later as unintentional error conditions are
+discovered, but are not within scope at this time.
+
+Since an adequate starting point for gRPC with mTLS is provided as an example in
+the
+[`grpc-go` repository,](https://github.com/grpc/grpc-go/tree/master/examples/features/encryption/mTLS)
+that will be used as an initial base for the server and client implementations
+to grow out of.
+
+## Usage Examples
+
+### Environment Variables
+
+#### Server
+```sh
+TP_WORKER_IP=127.0.0.1                          # --ip-addr
+TP_WORKER_PORT=8080                             # --port
+TP_WORKER_USER_CERTDIR=/path/to/cert/dir/       # --user-certs
+TP_WORKER_PRIVATE_KEY=/path/to/private/key.pem  # --private-key
+```
+
+#### Client
+```sh
+TP_WORKER_IP=127.0.0.1                                # --ip-addr
+TP_WORKER_PORT=8080                                   # --port
+TP_WORKER_CLIENT_PRIVATE_KEY=/path/to/private/key.pem # --private-key
+```
+
+### Start the Server
+
+Starting the server attaches to server log output unless sent to the background.
+
+```sh
+tp-worker-server
+...
+< Logged Output >
+...
+```
+
+### Client Pings Server
+
+```sh
+tp-worker-client ping
+Server OK at 127.0.0.1:8080
+```
+
+### Client Starts a Job
+`ls -lha` used as example.
+
+The `--name` argument can be passed to provide a unique name to the job,
+otherwise we assign the job a unique name based on the command and an increment.
+Names must start with a letter to prevent confusion with pid's elsewhere. Names
+are global. Fails if the name is already taken.
+
+The `-f` or `--follow` argument is equivalent to `tail -f`, which immediately
+attaches to the streaming output of the job.
```

Comment:
Would it simplify things to just make this the default behavior and not have an option?